back-end/lms/views/absence.py

```diff
@@ -116,44 +162,122 @@ def put(self, request: Request) -> Response:
         :param request:
         :return:
         """
-        absence = AbsenceSerializer(data=request.data)
-        try:
-            if absence.is_valid():
-                absence = absence.save()
+        existing_absence = Absence.objects.filter(id=request.data['id'])
```

Comment:
Consider passing `id` as part of the URL: `POST /api/lms/absence/{id}/`. 
This way you don't have to check for existence manually.